ecommerce/CheckoutController.php

```diff
@@ -619,16 +680,15 @@ public static function prepare_payment_data( array $order ) {
 					'discounted_price' => is_null( $item->discount_price ) || '' === $item->discount_price ? null : $item->discount_price,
 				);
 
-				if ( $plan_info && property_exists( $plan_info, 'enrollment_fee' ) && $plan_info->enrollment_fee > 0 ) {
-					$enrollment_item = array(
-						'item_id'          => 0,
-						'item_name'        => 'Enrollment Fee',
-						'regular_price'    => floatval( $plan_info->enrollment_fee ),
-						'quantity'         => 1,
-						'discounted_price' => null,
-					);
+				$enrollment_fee = OrderMetaModel::get_meta_value( $order_id, OrderModel::META_ENROLLMENT_FEE, true );
+				$trial_fee      = OrderMetaModel::get_meta_value( $order_id, OrderModel::META_TRIAL_FEE, true );
 
-					$items[] = $enrollment_item;
+				if ( $enrollment_fee ) {
```

Comment:
Same here

Reply:
Refactored existing code to be more reusable. Further improvements will be done later.